packages/worker/src/worker.ts

```diff
@@ -398,15 +457,19 @@ export class Worker {
     const evictionPromise = firstValueFrom(
       constructedWorker.evictionsSubject.pipe(
         // Ignore self-induced shutdowns, as they are a legit and we'll terminate normally.
-        filter((ev) => ev.evictJob != this.SELF_INDUCED_SHUTDOWN_EVICTION),
-        map((ev) => {
-          if (ev.evictJob.reason === EvictionReason.NONDETERMINISM) {
+        filter(
+          ({ evictJob }) =>
+            evictJob.message !== this.SELF_INDUCED_SHUTDOWN_EVICTION.message &&
+            evictJob.reason !== this.SELF_INDUCED_SHUTDOWN_EVICTION.reason
```

Comment:
Why this `&&`? Seems possibly over-strict?

Reply:
What's the right logic here?

Reply:
This was something you originally implemented and I overlooked in the review, `==` doesn't work for deep equality check.

Reply:
Oh I didn't realize it was comparing against a object before. This is fine then -- better would be a `equals` method on the fake eviction thing or a `deepEqual` call or something like that. Seeing the fields fanned out here instead of that being encapsulated just makes it a bit harder to read/fragile IMO